File: hw1/src/test/java/ParserTest.java
import org.junit.Assert;
import org.junit.Test;

import java.util.InvalidPropertiesFormatException;

public class ParserTest {

    @Test
    public void parseOperatorTestPlus() throws InvalidPropertiesFormatException {
        String plus = Parser.parseOperatorOrError("+");
        Assert.assertEquals("+", plus);
    }

    @Test
    public void parseOperatorTestMinus() throws InvalidPropertiesFormatException {
        String minus = Parser.parseOperatorOrError("-");
        Assert.assertEquals("-", minus);
    }

    @Test
    public void parseOperatorTestMultiply() throws InvalidPropertiesFormatException {
        String multiply = Parser.parseOperatorOrError("*");
        Assert.assertEquals("*", multiply);
    }

    @Test
    public void parseOperatorTestDivide() throws InvalidPropertiesFormatException {
        String div = Parser.parseOperatorOrError("/");
        Assert.assertEquals("/", div);
    }


    @Test
    public void parseOperatorTestError() {
        Assert.assertThrows(InvalidPropertiesFormatException.class, () -> Parser.parseOperatorOrError("d"));
    }

    @Test
    public void parseIntTestSuccess() {
        int num = Parser.parseIntOrError("555");
        Assert.assertEquals(555, num);
    }

    @Test
    public void parseIntTestError() {
        Assert.assertThrows(NumberFormatException.class, () -> Parser.parseIntOrError("five"));
    }

    @Test
    public void countOfArgumentsCheckerTest() {
        Assert.assertThrows(InvalidPropertiesFormatException.class, () -> Parser.countOfArgumentsChecker(new String[4]));
    }
}

File: hw2/src/test/kotlin/ParserTest.kt
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

class ParserTest {

    private val parser = Parser()

    @Test
    fun parseOperatorTestPlus() {
        val plus: String = parser.parseOperatorOrError("+")
        Assertions.assertEquals("+", plus)
    }

    @Test
    fun parseOperatorTestMinus() {
        val minus: String = parser.parseOperatorOrError("-")
        Assertions.assertEquals("-", minus)
    }

    @Test
    fun parseOperatorTestMultiply() {
        val multiply: String = parser.parseOperatorOrError("*")
        Assertions.assertEquals("*", multiply)
    }

    @Test
    fun parseOperatorTestDivide() {
        val div: String = parser.parseOperatorOrError("/")
        Assertions.assertEquals("/", div)
    }


    @Test
    fun parseOperatorTestError() {
        Assertions.assertThrows(InvalidPropertiesFormatException::class.java) { parser.parseOperatorOrError("d") }
    }

    @Test
    fun parseIntTestSuccess() {
        val num: Int = parser.parseIntOrError("555")
        Assertions.assertEquals(555, num)
    }

    @Test
    fun parseIntTestError() {
        Assertions.assertThrows(NumberFormatException::class.java) { parser.parseIntOrError("five") }
    }

    @Test
    fun countOfArgumentsCheckerTest() {
        Assertions.assertThrows(InvalidPropertiesFormatException::class.java) {
            parser.countOfArgumentsChecker(arrayOf("5","7"))
        }
    }
}